lib/rucio/core/rule.py

```diff
@@ -2414,22 +2414,24 @@ def examine_rule(rule_id: str, *, session: "Session") -> dict[str, Any]:
 
 
 @transactional_session
-def get_evaluation_backlog(expiration_time: int = 600, *, session: "Session") -> tuple[int, datetime]:
+def get_evaluation_backlog(expiration_time: int = 600, *, session: "Session") -> Union[NoValue, tuple[int, datetime]]:
```

Comment:
Under what circumstances can this function return `NoValue`?

Reply:
You're right, it doesn't happen under any circumstances.

The type checker complains when `if cached_backlog is NO_VALUE:` is used as the condition - it still thinks that `cached_backlog` might be of type `NoValue` outside of the if statement. If I use `if isinstance(cached_backlog, NoValue):` instead, it works correctly, but the `NoValue` descriptor mentions that `The .NO_VALUE constant should be used.` What do you think?

Reply:
Good catch, `isinstance(cached_backlog, NoValue)` is the correct way to implement a type guard.